bench/app.py

```diff
@@ -39,14 +39,22 @@
 
 
 class AppMeta:
-	def __init__(self, name: str, branch: str = None, to_clone: bool = True):
+	def __init__(
+		self,
+		name: str,
+		branch: str = None,
+		git_host: str = None,
+		to_clone: bool = True,
+		use_ssh: bool = False
+	):
 		"""
 		name (str): This could look something like
 			1. https://github.com/frappe/healthcare.git
 			2. git@github.com:frappe/healthcare.git
 			3. frappe/healthcare@develop
-			4. healthcare
-			5. healthcare@develop, healthcare@v13.12.1
+			4. frappe/healthcare
+			5. healthcare
+			6. healthcare@develop, healthcare@v13.12.1
```

Comment:
I think we can add support for `github:frappe/healthcare@version-13` too? SInce we want the gitserver host to be available too 🤔 . We could allow passing TLDs too instead of hardcoding them unconditionally in the `get_*_url` methods. Eg: `git.gav.in:frappe/healthcare@version-13` would also be valid.

Reply:
tbh i'm not sure ..i was thinking of removing the one i've added as it seemed a bit excessive to me (kinda went with the flow 😅 ) - though if you want it i can add that but the host one is kinda solved using `--git-host` option

Reply:
Passing `--git-host` is one solution that seems to miss out on the whole idea of why this format was added. 

We should get rid of the git_host kwarg altogether. Just fetch this by parsing the url. That's how partial URLs should be handled. Adding a new option doesn't work while integrating this system in the App Dependency System.

Reply:
hmm, so basically the proposal s to get the host from the passed format itself rather than providing it through an option (?) - we can do that though i'm not sure how much different they are though cause they basically achieve the same thing 🤔

Reply:
> I think we can add support for `github:frappe/healthcare@version-13` too? SInce we want the gitserver host to be available too 🤔 . We could allow passing TLDs too instead of hardcoding them unconditionally in the `get_*_url` methods. Eg: `git.gav.in:frappe/healthcare@version-13` would also be valid.

If we end up supporting hosts in that partial fragment too, I think at that point we might as well expect people to use the full URL anyway, no?

The new `get-app` is a good option to use when some of the things expected in the URL can be defaulted (host to github, ref to default branch/tag). IMHO, anything beyond that should be received either in the full URL, or as separate options.

Reply:
+1 for alchez's comment - i think most people don't really care about shortcuts like these - we/they just go to a repo and get the git url and start doing things